[Timothy Leow] iP

timleow opened the pull request:
# WallE 🤖
> “Humans would put seeds in the ground, pour water on them, and they grow food—like, pizza.”

WallE is a to-do list that is:
- Text-based
- ~~Smart~~ **Super Smart**
- Straight out of Disney 👑 

WallE will remember your
1. ToDo items
2. Deadlines
3. Events

What we've planned for WallE so far: 
- [x] Intuitive text commands
- [ ] Logical responses _with error handling_
- [ ] Intuitive GUI

Here's an example `command` that WallE can execute!
```java
public class ListTasksCommand implements Command {
    /**
     * {@inheritDoc}
     * Executes the command to list out all the tasks that are currently stored in the TaskList.
     */
    @Override
    public void execute(TaskList tasks, Ui ui, Storage storage) {
        for (int i = 0; i < tasks.getSize(); i++) {
            Task task = tasks.getTask(i);
            ui.printTask(i, task);
        }
    }
}
```
Check out the first release [here](https://github.com/timleow/ip/releases/tag/A-Jar)!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

timleow committed:
First commit, level 0

timleow committed:
Level-1 Bot, echoes input until bye is said

timleow committed:
Level 2 bot: saves input to a list that can be printed out on request

timleow committed:
Can mark tasks as done and undone

timleow committed:
Events, Deadlines and Tasks are now implemented

timleow committed:
I/O testing

timleow committed:
Error handling

timleow committed:
Implemented delete

timleow committed:
Implement saving and loading of tasks

timleow committed:
Implement deadline LocalDate and LocalDateTime processing

timleow committed:
Refactor code for more OOP

timleow committed:
Move java files into packages

timleow committed:
Merge branch 'master' into add-gradle-support

* master:
  Move java files into packages
  Refactor code for more OOP
  Implement deadline LocalDate and LocalDateTime processing
  Implement saving and loading of tasks
  Implemented delete
  Error handling
  I/O testing
  Events, Deadlines and Tasks are now implemented
  Can mark tasks as done and undone
  Level 2 bot: saves input to a list that can be printed out on request
  Level-1 Bot, echoes input until bye is said
  First commit, level 0

timleow committed:
Merge master (up to A-MoreOOP) into add-gradle-support

timleow committed:
Add gradle support to master

timleow committed:
Add Junit tests for Storage and Parser

timleow committed:
Add some javadocs

timleow committed:
Add JavaDocs for Commands, Duke, OOP packages

timleow committed:
Implement find task command

timleow committed:
Add JavaDocs for Tasks pacakge and Ui file in OOP package

timleow committed:
Modify code to adhere to coding standard and add checkstyle functionality

wesho1107 commented on src/main/java/Commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,23 @@
+import Tasks.Deadline;
+import Tasks.Task;
+public class AddDeadlineCommand implements Command {
+    private Deadline deadline;
```
I'm not sure if I like the variable declaration and functions not being separated by one blank line.

Do you think logical units within a block should be separated by one blank line?

wesho1107 commented on src/main/java/OOP/Parser.java:
```diff
@@ -0,0 +1,97 @@
+                        case "event":
+                            String[] threeParts = userCommandText.split ("/");
+                            String eventName = extractSecondWordOnwards(threeParts[0]);
+                            if (eventName.length() == 0) {
+                                throw new DukeException("\tEmpty Description");
+                            }
+                            if (threeParts.length != 3) {
+                                throw new DukeException("\tIncorrect format for event." +
+                                                        "\n\tExpected usage: " +
+                                                        "event {eventName} /from {eventStart} /to {eventEnd}");
+                            }
+                            String eventStart = extractSecondWordOnwards(threeParts[1]);
+                            String eventEnd = extractSecondWordOnwards(threeParts[2]);
+                            if (eventStart.length() == 0 || eventEnd.length() == 0) {
+                                throw new DukeException("\tBoth event start and end date times must be specified.");
+                            }
+                            return new AddEventCommand(eventName, eventStart, eventEnd);
+                        case "mark":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: mark {id}");
+                            }
+                            return new MarkTaskCommand(id);
+                        case "unmark":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: unmark {id}");
+                            }
+                            return new UnmarkTaskCommand(id);
+                        case "delete":
+                            try {
+                                id = Integer.valueOf(inputWords[1]) - 1;
+                            } catch (RuntimeException e) {
+                                throw new DukeException("\tExpected usage: delete {id}");
+                            }
+                            return new DeleteTaskCommand(id);
+                        default:
+                            return new InvalidCommand();
```
Should the case statements be indented according to Java coding standard introduced by the module?

wesho1107 commented on src/test/java/OOP/StorageTest.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import Tasks.Task;
```
Any reason why the Tasks.task import is standalone?

wesho1107 commented on src/main/java/Commands/AddEventCommand.java:
```diff
@@ -0,0 +1,23 @@
+import OOP.Ui;
+import OOP.Storage;
+import Tasks.Task;
+import Tasks.Event;
```
Should the imports and main class body be separated by a blank line?

wesho1107 reviewed (commented):
I like how your classes were split, it was very clear when looking for links between your code. 

I think the minor coding standard violations were mostly spacings between logical units in your code. Also, perhaps JavaDocs could be added for clarity of important methods and classes!

victorpengmx commented on src/main/java/OOP/Storage.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+
+    private static boolean isValidDate(String input) {
```
Should there be multiple blank lines between lines of code?

victorpengmx commented on src/main/java/OOP/Parser.java:
```diff
@@ -0,0 +1,97 @@
+import Duke.DukeException;
+
+public class Parser {
+    public static Command parseCommand(String userCommandText) {
+            if (!userCommandText.equals("bye")) {
+                if (!userCommandText.equals("list")) {
+                    String[] inputWords = userCommandText.split(" ");
+                    int id;
+                    switch (inputWords[0]) {
+                        case "todo":
+                            String todoName = extractSecondWordOnwards(userCommandText);
+                            if (todoName.length() == 0) {
+                                throw new DukeException("\tEmpty Description");
+                            }
+                            return new AddToDoCommand(todoName);
+                        case "deadline":
+                            String[] twoParts = userCommandText.split ("/by ");
+                            String deadlineName = extractSecondWordOnwards(twoParts[0]);
+                            if (deadlineName.length() == 0) {
+                                throw new DukeException("\tEmpty Description");
+                            } else if (twoParts.length != 2) {
+                                throw new DukeException("\tUsage: deadline {taskName} /by {yyyy-MM-dd HHmm}");
+                            }
+                            String deadlineString = twoParts[1];
+                            return new AddDeadlineCommand(deadlineName, deadlineString);
+                        case "event":
+                            String[] threeParts = userCommandText.split ("/");
+                            String eventName = extractSecondWordOnwards(threeParts[0]);
+                            if (eventName.length() == 0) {
+                                throw new DukeException("\tEmpty Description");
+                            }
+                            if (threeParts.length != 3) {
+                                throw new DukeException("\tIncorrect format for event." +
+                                                        "\n\tExpected usage: " +
+                                                        "event {eventName} /from {eventStart} /to {eventEnd}");
+                            }
+                            String eventStart = extractSecondWordOnwards(threeParts[1]);
+                            String eventEnd = extractSecondWordOnwards(threeParts[2]);
+                            if (eventStart.length() == 0 || eventEnd.length() == 0) {
+                                throw new DukeException("\tBoth event start and end date times must be specified.");
```
Perhaps avoid methods more better readibility?

victorpengmx commented on src/main/java/Commands/AddToDoCommand.java:
```diff
@@ -0,0 +1,22 @@
+import OOP.Storage;
+import Tasks.Task;
+import Tasks.ToDo;
+public class AddToDoCommand implements Command {
```
I'm not sure I like the import groups and class not separated by blank lines.

victorpengmx commented on src/main/java/OOP/Parser.java:
```diff
@@ -0,0 +1,97 @@
+        String[] wordArray = str.split(" ");
+        String secondWordOnwards = wordArray.length >= 2 ? wordArray[1] : "";
+        for (int i = 2; i < wordArray.length; i++)  {
+            secondWordOnwards += " " + wordArray[i];
```
I like the use of whitespace around operators. Makes the code look neater and more readable.

victorpengmx reviewed (commented):
Apart from a few nits, LGTM!

timleow committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDocs for Tasks pacakge and Ui file in OOP package
  Add JavaDocs for Commands, Duke, OOP packages

timleow committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Modify code to adhere to coding standard and add checkstyle functionality

timleow committed:
Merge level-9 to master

timleow committed:
Implement CheckStyle in accordance with 2103 guidelines

timleow committed:
Implement graphical GUI with wallE :)

timleow committed:
Implement java assertions

timleow committed:
Implement FindTasksCommand using streams

timleow committed:
Add Code Quality Improvements.

In Parser.java, the happy path is nested, we use guard clauses to make
it clearer and more prominent.
There are also multiple levels of abstraction within the switch
statements, which make the method parseCommand excessively long.
As such, we abstract out string processing statements
into smaller and shorter methods.

In Ui.java,
* We remove dead code that print to the terminal since the text UI is
no longer used.
* We avoid magic strings for the farewell text as it is used to check
for an exit command in MainWindow.java

We also make other improvements and minor adjustments to tests and
commands, and delete dead code in this refactoring.

timleow committed:
Merge pull request #3 from timleow/branch-A-CodeQuality

Add Code Quality Improvements.

timleow committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Add Code Quality Improvements.

# Conflicts:
#	src/main/java/Commands/DeleteTaskCommand.java
#	src/main/java/Commands/FindTasksCommand.java
#	src/main/java/Commands/MarkTaskCommand.java
#	src/main/java/Commands/UnmarkTaskCommand.java
#	src/main/java/OOP/Parser.java

timleow committed:
Merge pull request #1 from timleow/branch-A-Assertions

Implement java assertions

timleow committed:
Merge branch 'master' into branch-A-Streams

* master:
  Add Code Quality Improvements.
  Implement java assertions

# Conflicts:
#	src/main/java/Commands/FindTasksCommand.java

timleow committed:
Merge pull request #2 from timleow/branch-A-Streams

Implement FindTasksCommand using streams

timleow committed:
Implement Help Pop-up

It is difficult for the user to use wallE without knowing all the commands. Hardcoding the description in the fxml file is the fastest option to get this to feature up for now.

Future consideration is to have all the command formats and descriptions saved by Duke, and then rendered through a loop instead of hardcoding in the fxml file.

timleow committed:
Merge pull request #4 from timleow/branch-C-Help

Implement Help Pop-up

timleow committed:
Implement CI, without I/O text-ui testing

timleow committed:
Revert "Implement CI, without I/O text-ui testing"

This reverts commit 9e4033a9cf3a4b68db4273702ec39d7c58a6c925.

timleow committed:
Revert "Revert "Implement CI, without I/O text-ui testing""

This reverts commit dad15d31a334a29fe3c47622f620fbbb4377de24.

timleow committed:
Undo adding of yml file due to PAT scope limit

timleow committed:
Create gradle.yml for CI

timleow committed:
Add User Guide and other minor fixes

timleow committed:
Fix style and other minor changes

timleow committed:
Correct build.gradle with the right Main file duke.Duke

timleow committed:
Update README.md

timleow committed:
Update README.md

timleow committed:
Update README.md

timleow committed:
Update README.md
